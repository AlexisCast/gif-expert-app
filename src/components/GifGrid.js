import React, { useEffect, useState } from "react";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
	const [images, setImages] = useState([]);
	useEffect(() => {
		GetGifs();
	}, []);
	const GetGifs = async () => {
		const url =
			"http://api.giphy.com/v1/gifs/search?q=Naruto&limit=10&api_key=oBJaWvD8oC03bP7ULhrHGNOkar6Jns6Y";
		const resp = await fetch(url);
		const { data } = await resp.json();
		const gifs = data.map((img) => {
			return {
				id: img.id,
				title: img.title,
				url: img.images?.downsized_medium.url,
			};
		});
		// console.log(data);
		setImages(gifs);
		console.log(gifs);
	};
	// GetGifs();
	return (
		<div>
			<h3>{category}</h3>

			{images.map((img) => (
				<GifGridItem key={img.key} {...img} />
			))}
		</div>
	);
};
