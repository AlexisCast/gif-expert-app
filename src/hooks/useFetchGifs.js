import { useEffect, useState } from "react";
import { GetGifs } from "../helpers/getGifs";

export const useFetchGifs = (category ) => {
	const [state, setState] = useState({
		data: [],
		loading: true,
	});
	useEffect(() => {
		GetGifs(category).then((imgs) => {
         console.log('test',imgs)
			// setTimeout(() => {
				setState({
					data: imgs,
					loading: false,
				});
			// }, 3000);
		});
	}, [category]);
	return state; //{data:[],loading:true}
};
