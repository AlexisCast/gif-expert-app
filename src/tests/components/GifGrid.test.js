import React from "react";
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid/>',()=>{
   const category='Goku';
   test('debe de mostrar el componente correctamente',()=>{
      useFetchGifs.mockReturnValue({
         data:[],
         loading:true,
      });
      const wrapper=shallow(<GifGrid category={category}/>);
      expect(wrapper).toMatchSnapshot();
   });
   test('debe de mostrar items cuando se carga imagnes useFetchGifs',()=>{
      const gifs=[{
         id:'abc',
         url:'https://google.com',
         title:'cualquier cosa',
      },
      {
         id:'123',
         url:'https://maps.com',
         title:'maps',
      },]
      useFetchGifs.mockReturnValue({
         data:gifs,
         loading:false,
      });
      const wrapper=shallow(<GifGrid category={category}/>);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('p').exists()).toBe(false);
      expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
   })
})


