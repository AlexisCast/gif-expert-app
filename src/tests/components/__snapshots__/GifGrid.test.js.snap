// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid/> debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid/> debe de mostrar items cuando se carga imagnes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Goku
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="cualquier cosa"
      url="https://google.com"
    />
    <GifGridItem
      id="123"
      key="123"
      title="maps"
      url="https://maps.com"
    />
  </div>
</Fragment>
`;
